### ISSUE TITLE
Project generated in Xcode 6.3 crashes on running in the simulator

### ISSUE THREAD

**Issue by sx00** [John-Lluch/SWRevealViewController#402]
If I generate my project in Xcode 6.3 and run it in the simulator, I get this error (screenshot). There are no errors if the project generated in Xcode 6.2 or run on the iPhone.
![screen shot 2015-04-10 at 16 21 50](https://cloud.githubusercontent.com/assets/2961856/7090232/d63bc5fc-dfa1-11e4-9754-d8e59cc17700.png)

**Comment by kocyigityunus** [comment]
just add a new empty view controller. and connect it with swrevealviewcontrollerset segue named "sw_right"

**Comment by TheGhark** [comment]
Doing so breaks the purpose of having a @try block. Instead, you should include these lines:

    if ([self canPerformSegueWithIdentifier:SWSegueRightIndetifier]) {
        [self performSegueWithIdentifier:@"SegueID"];
    }

**Comment by elsesiy** [comment]
@kocyigityunus why would he want to do that? Why not make it configurable, so one could use the segues one want? He could override the method as specified in the comment above though, but therefore he'd need to subclass or manipulate the provided code which is both annoying IMHO

**Comment by eviltofu** [comment]
I have the same error BUT on hardware (iPad4,iPhone4S,iPhone5,iPhone6) it does not crash.

**Comment by elsesiy** [comment]
Same for me, app crashes only on iPad referencing the missing segue

**Comment by kocyigityunus** [comment]
@elsesiy i was in rush. I think this is happening because genius apple is not based nsinvalidargumentexception from nsexception as you can see there :  https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Miscellaneous/Foundation_Constants/index.html#//apple_ref/doc/constant_group/General_Exception_Names

We need to catch nsinvalidargumentexception also. I can try this and provide a gist soon.

**Comment by eviltofu** [comment]
Ok, I've created a simple test case which tries to throw an NSInvalidArgumentException but all exceptions thrown are NSExceptions. How do I generate an NSInvalidArgumentException? I've tried performing segues which do not exist (which clearly is the case here).

OS X 10.10.3, Xcode 6.3 (6D570), iOS 8.3 (simulator).

https://github.com/eviltofu/NSInvalidArgumentExceptionTest

**Comment by eviltofu** [comment]
If you add in an empty view controller to "sw_right", the app will now open an empty window when you swipe to the left.

**Comment by tdroenner** [comment]
Update to XCode 6.3.1 the bug isn't in simulated devices of iphone 5s and above. this is a apple simulator bug.

**Comment by sx00** [comment]
Yes, it's simulator bug, but even for Xcode 6.3.1 the bug is still here

**Comment by tdroenner** [comment]
Like I said. Choose iPhone 5s and later devices for simluator testing. the
bug isn't there. Only iPhone5 and lower.

2015-04-23 10:06 GMT+02:00 sx <notifications@github.com>:

**Comment by sx00** [comment]
![screen shot 2015-04-23 at 10 27 21](https://cloud.githubusercontent.com/assets/2961856/7293255/6a7f7686-e9a3-11e4-89c2-37ad528983c5.png)

**Comment by kocyigityunus** [comment]
just comment it out if you're not using it. it's a bug from apple.

**Comment by sx00** [comment]
I can't comment it cause I use this segue with another view controller. 
At this moment I just added empty view controllers.

**Comment by kocyigityunus** [comment]
if you use  swrevealcontroller's pan gesture you can pan to the empty view controller. you better check if this can happen to you.

**Comment by ashokdy** [comment]
No need to create a new empty view controller just change the segue  as custom and give segue class as swrevealcontrollersetcontroller because till xcode6.4 (my latest xcode version) by default the segue is show which won't be supported for swrevealcontrollersetcontroller.... Happy coding. I hope it helped some one like me
